Additional functions for win_iis module & state

Mapel88 opened the pull request:
### What does this PR do?
Updates existing web application

### What issues does this PR fix or reference?
No open requests found

### Tests written?
No


State set_app usage:
```   
    '''
    Set the value of the setting for an IIS web application.
    .. note:
        This function only configures existing app
    :param str name: The IIS application.
    :param str site: The IIS site name.
    :param str settings: A dictionary of the setting names and their values.
    :available settings:    physicalPath: The physical path of the webapp.
    :                       applicationPool: The application pool for the webapp.
    :                       userName: "connectAs" user
    :                       password: "connectAs" password for user
    :rtype: bool
    Example of usage:

    .. code-block:: yaml

        site0-webapp-setting:
            win_iisV2.set_app:
                - name: app0
                - site: Default Web Site
                - settings:
                    userName: domain\user
                    password: pass
                    physicalPath: c:\inetpub\wwwroot
                    applicationPool: appPool0
```
Module function get_webapp_settings usage:
```
    '''
    Get the value of the setting for the IIS web application.
    .. note:
        Params are case sensitive are case sensitive
    :param str name: The name of the IIS web application.
    :param str site: The site name contains the web application.
        Example: Default web site
    :param str settings: A dictionary of the setting names and their values.
        Available settings: physicalPath, applicationPool, userName, password
    :return: A dictionary of the provided settings and their values.
    Returns:
        dict: A dictionary of the provided settings and their values.
    .. versionadded:: 2016.11.0
    CLI Example:
    .. code-block:: bash
        salt '*' win_iisV2.get_webapp_settings name='app0' site='Default Web Site'
            settings="['physicalPath','applicationPool']"
    '''
```
Module function get_webapp_settings usage:
```
    '''
    Configure an IIS application.
    .. note:
        This function only configures existing app
    :param str name: The IIS application.
    :param str site: The IIS site name.
    :param str settings: A dictionary of the setting names and their values.
    :available settings:    physicalPath: The physical path of the webapp.
    :                       applicationPool: The application pool for the webapp.
    :                       userName: "connectAs" user
    :                       password: "connectAs" password for user
    :return: A boolean representing whether all changes succeeded.
    :rtype: bool
    CLI Example:
    .. code-block:: bash
        salt '*' win_iis.set_webapp_settings name='app0' site='site0' settings="{'physicalPath': 'C:\site0', 'apppool': 'site0'}"
    '''
```
Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

Mapel88 committed:
Update win_iis.py

Added functions get_webapp_settings set_webapp_settings for updating existing web application

Mapel88 committed:
Update win_iis.py

fixed remarks

Mapel88 committed:
Update win_iis.py

Added set_app function for updating existing web application

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+def get_webapp_settings(name, site, settings):
+    '''
+    Get the value of the setting for the IIS web application.
+    .. note:
```
This needs to be `.. note::`

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+    :return: A dictionary of the provided settings and their values.
+    Returns:
+        dict: A dictionary of the provided settings and their values.
+    .. versionadded:: 2016.11.0
```
This should probably be `.. versionadded:: Nitrogen`

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+    return ret
+
+
+
```
Only 2 lines between functions (this should be caught by the linter)

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+def set_webapp_settings(name, site, settings):
+    '''
+    Configure an IIS application.
+    .. note:
```
again: `.. note::`

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+            return False
+
+
+
```
Too many blank lines here (lint error). Should have at the most one blank line within a function.

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+    current_settings = get_webapp_settings(
+        name=name, site=site, settings=settings.keys())
+
+
```
Too many blank lines...

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+        except ValueError:
+            value = "'{0}'".format(settings[setting])
+
+
```
Too many blank lines...

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+            pscmd.append(" Set-WebConfigurationProperty -Filter \"system.applicationHost/sites/site[@name='{0}']/application[@path='/{1}']/virtualDirectory[@path='/']\"".format(site, name))
+            pscmd.append(" -Name \"{0}\" -Value {1};".format(setting, value))
+
+
```
Too many blank lines...

twangboy commented on salt/states/win_iis.py:
```diff
@@ -768,3 +768,82 @@ def remove_vdir(name, site, app='/'):
 
     return ret
+
+def set_app(name, site, settings=None):
```
Needs to be two blank lines between functions...

twangboy commented on salt/states/win_iis.py:
```diff
@@ -768,3 +768,82 @@ def remove_vdir(name, site, app='/'):
+def set_app(name, site, settings=None):
+    '''
+    Set the value of the setting for an IIS web application.
+    .. note:
```
`.. note::`

twangboy commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,180 @@ def list_worker_processes(apppool):
+    CLI Example:
+    .. code-block:: bash
+        salt '*' win_iis.set_webapp_settings name='app0' site='site0' settings="{'physicalPath': 'C:\site0', 'apppool': 'site0'}"
+    '''
```
This docstring needs a `.. versionadded:: Nitrogen` line

twangboy commented on salt/states/win_iis.py:
```diff
@@ -768,3 +768,82 @@ def remove_vdir(name, site, app='/'):
+                    password: pass
+                    physicalPath: c:\inetpub\wwwroot
+                    applicationPool: appPool0
+    '''
```
This docstring needs a `.. versionadded:: Nitrogen` line

twangboy reviewed (commented):
Other than some Lint errors and some missing docs stuff, it looks great.

Mapel88 committed:
Update win_iis.py

-Fixed blank lines.
-Fixed note: to note::
-Fixed versionadded:: 2016.11.0 to versionadded:: Nitrogen
-Fixed "Params are case sensitive. are case sensitive" to "Params are case sensitive"
-Fixed to 2 lines between functions
-Added note to set_webapp_settings "Params are case sensitive."
-Changed from settings[setting] to value

This is my first contribution ever and I don't use linter, so I hope I fixed all standardization issues you mentioned.

Mapel88 committed:
Update win_iis.py

-Fixed note: to note::
-Fixed 2 lines between functions
-Added .. versionadded:: Nitrogen
-Added "Params are case sensitive." to note

cachedout commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,173 @@ def list_worker_processes(apppool):
+    except ValueError:
+        log.error('Unable to parse return data as Json.')
+
+    if None in ret.viewvalues():
```
We need a `from salt.ext import six` added to the import section near the top of this file and then this line should be changed to `six.viewvalues(ret)`.

Mapel88 replied:
fixed

cachedout commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,173 @@ def list_worker_processes(apppool):
+    '''
+    pscmd = list()
+    current_apps = list_apps(site)
+    current_sites= list_sites()
```
One space before the `=` sign please.

Mapel88 replied:
fixed

cachedout commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,173 @@ def list_worker_processes(apppool):
+
+
+def set_webapp_settings(name, site, settings):
+    '''
```
We need an `r` prefix here, so `r'''`.

Mapel88 replied:
fixed

cachedout commented on salt/modules/win_iis.py:
```diff
@@ -1774,3 +1774,173 @@ def list_worker_processes(apppool):
+            pscmd.append(" -Name \"{0}\" -Value {1};".format(setting, value))
+
+        if setting == "physicalPath" or setting == "applicationPool":
+            pscmd.append(" Set-ItemProperty \"IIS:\Sites\{0}\{1}\" -Name {2} -Value {3};".format(site, name, setting, value))
```
This string needs an `r` prefix right before the opening double-quote.

Mapel88 replied:
not sure where exactly.. for every double quote should be 'r' prefix?

cachedout commented on salt/states/win_iis.py:
```diff
@@ -768,3 +768,85 @@ def remove_vdir(name, site, app='/'):
+
+
+def set_app(name, site, settings=None):
+    '''
```
This line need an `r` prefix as well, please.

Mapel88 replied:
fixed

cachedout reviewed (changes requested):
We need a few more changes here, please. Thanks!

Mapel88 committed:
Update win_iis.py

- Added from salt.ext import six
- Changed ret.viewvalues() to six.viewvalues(ret)
- Added space current_sites= list_sites()
- Added 'r' prefix

rallytime commented:
Hi @Mapel88 - There are a couple of lint failures related to your change. Would you mind cleaning those up really quickly? Then I think we'll be able to get this in.

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/9438/violations/file/salt/states/win_iis.py/

Mapel88 committed:
Update win_iis.py

added 'r'

cachedout commented:
I linted these latest changes locally and they all pass so this looks ready to go.

falenn commented:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  curtis.l.bates@gmail.com
    Domain imwiz.com has exceeded the max emails per hour (171/150 (113%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o11.sgmail.github.com ([167.89.101.202]:13317)
	by box969.bluehost.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.87)
	(envelope-from <bounces+848413-5c7e-dev=imwiz.com@sgmail.github.com>)
	id 1cnrcd-0005I9-00
	for dev@imwiz.com; Tue, 14 Mar 2017 12:56:45 -0600
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=R+/X/HX9Th0L+ERLAsicLJuafZQ=; b=bIzsEVaGF4UGstrE
	MGrFNPidGfFflLVcEQtn54G9ump+QCgqoBk0hcLjGKWZZA2z6FEHRjWollgHx/5A
	8FKds6pXDr9TtwqmdH+jfQWn0VCdCtnqXRiQ2Kmy8XVkibORrtQS7evBtc1RZX1Z
	yHMq0791nj9q3NMyzB9Y/d44AB8=
Received: by filter0470p1mdw1.sendgrid.net with SMTP id filter0470p1mdw1-25551-58C83CC4-99
        2017-03-14 18:56:04.963028404 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0003p1iad1.sendgrid.net (SG) with ESMTP id T1ftVIApTZuMQEVGTWi0uw
	for <dev@imwiz.com>; Tue, 14 Mar 2017 18:56:04.971 +0000 (UTC)
Date: Tue, 14 Mar 2017 11:56:04 -0700
From: Mike Place <notifications@github.com>
Reply-To: saltstack/salt <reply@reply.github.com>
To: saltstack/salt <salt@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Message-ID: <saltstack/salt/pull/39886/c286524921@github.com>
In-Reply-To: <saltstack/salt/pull/39886@github.com>
References: <saltstack/salt/pull/39886@github.com>
Subject: Re: [saltstack/salt] Additional functions for win_iis module & state
 (#39886)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_58c83cc4b6474_25bf3f94774e9c38191677";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: cachedout
X-GitHub-Recipient: falenn
X-GitHub-Reason: subscribed
List-ID: saltstack/salt <salt.saltstack.github.com>
List-Archive: https://github.com/saltstack/salt
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+000ab60afb14b01d254a193547389f46c7890965e1a490b292cf0000000114dffec492a169ce0cadf37c@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AAq2CnFDiYjeco73AtiD0knvpHbQjJ1Cks5rluLEgaJpZM4MWv5e>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: dev@imwiz.com
X-SG-EID: APO41b8ovafPb3SK9rw3vERqfAVL1M44P2sE3X/pAHFoZZs9GBF8ijbUZFnEMTf/bLODNrgM4UD6+D
 tCHLMX5RO3wEt4HL5rn8kikiT6KxfLZX+N6Osky2Ab891x5JmHbOihGV3cpU54bGM/awI/IwQSa/gX
 +e+qR3ZCvo/Fa310lgELoDxbP0SqPQRzHbRBWmlw30D2Fvaizxw2YIkBaQ==
X-Spam-Status: No, score=0.1
X-Spam-Score: 1
X-Spam-Bar: /
X-Ham-Report: Spam detection software, running on the system "box969.bluehost.com",
 has NOT identified this incoming email as spam.  The original
 message has been attached to this so you can view it or label
 similar future email.  If you have any questions, see
 root\@localhost for details.
 
 Content preview:  I linted these latest changes locally and they all pass so
    this looks ready to go. -- You are receiving this because you are subscribed
    to this thread. Reply to this email directly or view it on GitHub: https://github.com/saltstack/salt/pull/39886#issuecomment-286524921
    [...] 
 
 Content analysis details:   (0.1 points, 4.0 required)
 
  pts rule name              description
 ---- ---------------------- --------------------------------------------------
 -0.0 RCVD_IN_MSPIKE_H4      RBL: Very Good reputation (+4)
                             [167.89.101.202 listed in wl.mailspike.net]
 -0.0 SPF_PASS               SPF: sender matches SPF record
 -0.0 RP_MATCHES_RCVD        Envelope sender domain matches handover relay domain
  0.7 HTML_IMAGE_ONLY_20     BODY: HTML: images with 1600-2000 bytes of words
  0.0 HTML_MESSAGE           BODY: HTML included in message
 -0.1 DKIM_VALID             Message has at least one valid DKIM or DK signature
  0.1 DKIM_SIGNED            Message has a DKIM or DK signature, not necessarily valid
 -0.1 DKIM_VALID_AU          Message has a valid DKIM or DK signature from author's
                             domain
 -0.0 RCVD_IN_MSPIKE_WL      Mailspike good senders
 -0.5 AWL                    AWL: Adjusted score from AWL reputation of From: address
X-Spam-Flag: NO

----==_mimepart_58c83cc4b6474_25bf3f94774e9c38191677
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

I linted these latest changes locally and they all pass so this looks ready to go.

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/saltstack/salt/pull/39886#issuecomment-286524921
----==_mimepart_58c83cc4b6474_25bf3f94774e9c38191677
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>I linted these latest changes locally and they all pass so this looks ready to go.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br />You are receiving this because you are subscribed to this thread.<br />Reply to this email directly, <a href="https://github.com/saltstack/salt/pull/39886#issuecomment-286524921">view it on GitHub</a>, or <a href="https://github.com/notifications/unsubscribe-auth/AAq2Co1ePc0dRj0YNDa3HEfJf2H14IAUks5rluLEgaJpZM4MWv5e">mute the thread</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/AAq2CuS8Su77lJ08qWNFNoF3xBfppMLjks5rluLEgaJpZM4MWv5e.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/saltstack/salt/pull/39886#issuecomment-286524921"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

<script type="application/json" data-scope="inboxmarkup">{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/saltstack/salt","title":"saltstack/salt","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/saltstack/salt"}},"updates":{"snippets":[{"icon":"PERSON","message":"@cachedout in #39886: I linted these latest changes locally and they all pass so this looks ready to go."}],"action":{"name":"View Pull Request","url":"https://github.com/saltstack/salt/pull/39886#issuecomment-286524921"}}}</script>
----==_mimepart_58c83cc4b6474_25bf3f94774e9c38191677--
